### PR TITLE
contrib/net/http: don't set empty string values as span tags

### PR DESCRIPTION
**Pull request by knusbaum**
### What does this PR do?

Do not set span fields when they are not configured so the tracer can `put` the defaults in.

### Motivation

Currently the TraceAndServe function sets service, resource, etc. whether or not they are configured, meaning we can easily end up with spans with service "unnamed-go-service" and blank tags/resource. 

This is becoming relevant for the autoinstrumentation, where the service, etc is not readily available, and we want sane defaults.

### Reviewer's Checklist

- [ ] Changed code has unit tests for its functionality.
- [ ] If this interacts with the agent in a new way, a system test has been added.

### CODE REVIEW

File: contrib/net/http/trace_test.go [hunk 1]
```diff
@@ -288,6 +288,34 @@ func TestTraceAndServe(t *testing.T) {
...
+		}
+		TraceAndServe(http.HandlerFunc(handler), w, r, &ServeConfig{})
+		spans := mt.FinishedSpans()
+		span := spans[0]
+
+		assert.True(called)
+		assert.Len(spans, 1)
+		assert.Equal(ext.SpanTypeWeb, span.Tag(ext.SpanType))
+		assert.Nil(span.Tag(ext.ServiceName)) // This is nil since mocktracer does not behave like the actual tracer, which will set a default.
+		assert.Equal("http.server.request", span.Tag(ext.ResourceName))
```
Comment (ahmed-mez) [inline contrib/net/http/trace_test.go:314]: Is this also a _good_ default value that we want to keep?

File: contrib/net/http/trace_test.go [hunk 2]
```diff
@@ -288,6 +288,34 @@ func TestTraceAndServe(t *testing.T) {
...
+		assert.Equal("GET", span.Tag(ext.HTTPMethod))
+		assert.Equal("/path?<redacted>", span.Tag(ext.HTTPURL))
+		assert.Equal("200", span.Tag(ext.HTTPCode))
+	})
```
Comment (ahmed-mez) [inline contrib/net/http/trace_test.go:318]: I guess we can also assert that the route tag wasn't set to an empty value?

**Review by ahmed-mez — APPROVED** [review]

File: contrib/net/http/trace_test.go (hunk 1 above)
Comment (knusbaum) [reply contrib/net/http/trace_test.go:314]: I believe so, yes. This appears to be the new "ecosystems"-approved tag value.

**Review by ahmed-mez — APPROVED** [review]
:shipit:

**Pull request merged**